### PR TITLE
xmlstarlet: update to 1.6.7

### PR DESCRIPTION
**Pull request by heitbaum**
update 1.6.1 to 1.6.7
- no code changes apart from exit() to return()
- new PKG_SITE and PKG_URL
- improvements could be done by moving back to configure, issues with configure in the xmlstarlet directory not configuring correctly using the build toolchain

### CODE REVIEW

File: packages/textproc/xmlstarlet/package.mk [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-PKG_VERSION="1.6.1"
-PKG_SHA256="15d838c4f3375332fd95554619179b69e4ec91418a3a5296e7c631b7ed19e7ca"
+PKG_VERSION="1.6.7"
+PKG_SHA256="719f64b81a800778d15ce2984acf48d6c2044f8e51c75a021a5b94e87437b414"
 PKG_LICENSE="MIT"
-PKG_SITE="http://xmlstar.sourceforge.net"
-PKG_URL="http://netcologne.dl.sourceforge.net/project/xmlstar/$PKG_NAME/$PKG_VERSION/$PKG_NAME-$PKG_VERSION.tar.gz"
+PKG_SITE="https://pypi.org/project/xmlstarlet"
+PKG_URL="https://files.pythonhosted.org/packages/source/${PKG_NAME:0:1}/$PKG_NAME/$PKG_NAME-$PKG_VERSION.tar.gz"
+PKG_ID="31079"
```
Comment (CvH) [inline packages/textproc/xmlstarlet/package.mk:11]: :)

**Comment by vpeter4** [comment]
Something like this should work too:
```
PKG_TOOLCHAIN="configure"
PKG_CONFIGURE_SCRIPT="../xmlstarlet/configure"

post_unpack() {
  ( cd ${PKG_BUILD}/xmlstarlet && make distclean )
}

pre_configure_host() {
  mkdir -p ${PKG_BUILD}/.$HOST_NAME
  cd ${PKG_BUILD}/.$HOST_NAME
}

pre_configure_target() {
  mkdir -p ${PKG_BUILD}/.$TARGET_NAME
  cd ${PKG_BUILD}/.$TARGET_NAME
}
```

**Comment by heitbaum** [comment]
@vpeter4 - thanks for the tip. I’ll give it a go.

**Comment by vpeter4** [comment]
You can use even shorter solution.
```
PKG_CONFIGURE_SCRIPT="$BUILD/build/${PKG_NAME}-${PKG_VERSION}/xmlstarlet/configure"

post_unpack() {
  ( cd ${PKG_BUILD}/xmlstarlet && make distclean >/dev/null )
}
```

**Comment by vpeter4** [comment]
Btw, someone should tell xmlstarlet devs that they should not be ship source archive configured and with object files.

**Comment by mglae** [comment]
I know, I'm a little late with my 5ct :-)

We can get the pure source from https://github.com/dimitern/xmlstarlet
Drop the patch change and "normalize" the source: http://ix.io/2Pxu

**Comment by heitbaum** [comment]
@mglae - I have updated the PR and tested both x86 and Amlogic builds - both good. I think it is no longer WIP, but will need some testing across the builders now.

**Comment by vpeter4** [comment]
Moving files around are unnecessary. You can specify **PKG_CONFIGURE_SCRIPT** as I already suggested or use custom **unpack()** function.
```
unpack() {
  mkdir -p ${PKG_BUILD}
  tar --strip-components=2 -xf ${SOURCES}/${PKG_NAME}/${PKG_NAME}-${PKG_VERSION}.tar.gz -C ${PKG_BUILD} ${PKG_NAME}-${PKG_VERSION}/xmlstarlet
}
```

**Comment by mglae** [comment]
@vpeter4  Setting PKG_CONFIGURE_SCRIPT failed while testing. #5144 is needed for:

```
configure_package() {
  PKG_CONFIGURE_SCRIPT="${PKG_BUILD}/xmlstarlet/configure"
}
```

**Comment by vpeter4** [comment]
I tested PKG_CONFIGURE_SCRIPT change and worked last time. Maybe I did something wrong.
Anyway, there are different options to use than moving files around.

**Comment by mglae** [comment]
Before it did work because `configure` already was created in the downloaded source. From GitHub we only get `configure.ac` requiring autoreconf.

**Comment by lrusak** [comment]
the github link seems to provide python cffi bindings for xmlstarlet with xmlstarlet source as a subdir. Why not use the actual source from sourceforge?

**Comment by vpeter4** [comment]
But I see autoreconf called only with this PR.

**Comment by CvH** [comment]
just for info https://sourceforge.net/p/xmlstar/discussion/226077/thread/fbf4612d3f/
the current source (that pr) is basically a semi fork of xmlstarlet, the original is still at SF and is basically dead

**Comment by lrusak** [comment]
> just for info https://sourceforge.net/p/xmlstar/discussion/226077/thread/fbf4612d3f/
> the current source (that pr) is basically a semi fork of xmlstarlet, the original is still at SF and is basically dead

I don't think we need to use a fork.

arch: https://github.com/archlinux/svntogit-community/blob/packages/xmlstarlet/trunk/PKGBUILD#L16
fedora: https://src.fedoraproject.org/rpms/xmlstarlet/blob/rawhide/f/xmlstarlet.spec#_9

**Comment by vpeter4** [comment]
The point of this PR is update 1.6.1 to 1.6.7.

**Comment by lrusak** [comment]
> The point of this PR is update 1.6.1 to 1.6.7.

Then IMO the PR should be closed

**Comment by vpeter4** [comment]
But all the packages must be bumped :-)

**Comment by CvH** [comment]
closing for now

**Comment by heitbaum** [comment]
Agree

**Pull request closed without merging**